R/dash.R

```diff
@@ -374,55 +344,24 @@ Dash <- R6::R6Class(
     # ------------------------------------------------------------------------
     # callback registration
     # ------------------------------------------------------------------------
-    callback = function(func = NULL, output = NULL) {
+    callback = function(output = NULL, inputs = list(), state = list(), user_function) {
+      private$layout_render()
 
-      # argument type checking
-      assertthat::assert_that(is.function(func))
-      assertthat::assert_that(is.output(output))
-
-      # TODO: cache layouts so we don't have to do this for every callback...
-      layout <- private$layout_render()
-      if (identical(layout, welcome_page())) {
-        stop("The layout must be set before defining any callbacks", call. = FALSE)
-      }
+      assert_valid_callbacks(output, inputs, state, user_function)
 
-      # -----------------------------------------------------------------------
-      # verify that output/input/state IDs provided exists in the layout
-      # -----------------------------------------------------------------------
-      callbackInputs <- callback_inputs(func)
-      callbackStates <- callback_states(func)
+      # set class attributes for compatibility with original DashR code
```

Comment:
Can we delete this in the future? I don't think we should worry about backwards compatability until we officially release.

Reply:
Yes; we don't actually need the class attributes, but I can understand why Carson used them.